src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -1222,7 +1222,28 @@ namespace QgsWfs
           const bool invertAxis { mWfsParameters.versionAsNumber() >= QgsProjectVersion( 1, 1, 0 ) &&
                                   crs.hasAxisInverted() &&
                                   ! srsName.startsWith( QLatin1String( "EPSG:" ) ) };
-          QDomElement envElem = QgsOgcUtils::rectangleToGMLEnvelope( rect, doc, srsName, invertAxis, prec );
+
+          // If requested SRS (srsName) is different from rect CRS (crs) we need to transform the envelope
+          QgsRectangle crsCorrectedRect { rect ? *rect : QgsRectangle() };
+          QgsCoordinateTransform transform;
+          transform.setSourceCrs( crs );
+          transform.setDestinationCrs( QgsCoordinateReferenceSystem( srsName ) );
+          QgsGeometry exportGeom { QgsGeometry::fromRect( *rect ) };
+          try
+          {
+            if ( exportGeom.transform( transform ) == Qgis::GeometryOperationResult::Success )
```

Comment:
Usually we use transform.transformBoundingBox here -- that has better logic to handle densification of the original bounding box to ensure more reliable results

Reply:
Done, should I pass handle180Crossover true?

Reply:
No, you can leave that as false